docs/configure-artifact-repository.md

```diff
@@ -107,43 +111,181 @@ $ cat > policy.json <<EOF
    ]
 }
 EOF
-$ aws s3 mb s3://$mybucket [--region xxx]
-$ aws iam create-user --user-name $mybucket-user
-$ aws iam put-user-policy --user-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
-$ aws iam create-access-key --user-name $mybucket-user > access-key.json
 ```
 
 If you do not have Artifact Garbage Collection configured, you should remove `s3:DeleteObject` from the list of Actions above.
 
-NOTE: if you want argo to figure out which region your buckets belong in, you
-must additionally set the following statement policy. Otherwise, you must
-specify a bucket region in your workflow configuration.
+!!! Note "Discovering bucket regions"
+    If you want Argo Workflows to figure out which region your buckets belong in, you must additionally set the following statement policy.
+    Otherwise, you must specify a bucket region in your workflow configuration.
 
 ```json
       {
-         "Effect":"Allow",
-         "Action":[
+        "Effect":"Allow",
+        "Action":[
             "s3:GetBucketLocation"
-         ],
-         "Resource":"arn:aws:s3:::*"
+        ],
+        "Resource":"arn:aws:s3:::*"
       }
     ...
 ```
 
-### AWS S3 IRSA
+#### AWS S3 IRSA
+
+IAM Roles for Service Accounts (IRSA) is the recommended Kubernetes native mechanism to authenticate to S3.
+If you are using EKS, follow [the IRSA setup guide](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
+If not, follow the [Pod Identity Webhook self-hosted setup guide](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md).
 
-If you wish to use S3 IRSA instead of passing in an `accessKey` and `secretKey`, you need to annotate the service account of both the running workflow (in order to save logs/artifacts) and the argo-server pod (in order to retrieve the logs/artifacts).
+With the bucket and policy as described above, create an IAM role and add the policy:
+
+```bash
+aws iam create-role --role-name $mybucket-role
+aws iam put-role-policy --role-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+```
+
+To use S3 IRSA, you need to annotate the service account of both the running workflow (in order to save logs/artifacts) and the argo-server pod (in order to retrieve the logs/artifacts).
+You can attach only one IAM role to a service account, so your roles must have permission for all buckets you may wish to use.
```

Comment:
You didn't change this either, but this is also very confusing and suggests excessive privileges as well. In https://github.com/argoproj/argo-workflows/pull/13425#pullrequestreview-2214101470 I suggested separating out Server permissions into its own section, as it's optional and largely orthogonal given that it's read access vs write access.

For this PR, I might suggest simplifying this to only what the example below does, and leave the rest to a separate PR adding a Server section

```suggestion
Attach this IAM role to a service account with an annotation:
```

Reply:
Yep, can do. I feel like it is still important to have somewhere, since obviously when the artifacts are retrieved via the UI, it can fail, but agree, maybe it should be in the server docs rather than here.

Reply:
Yea I was thinking of adding a "## Server Read-only Access" section at the bottom of this page with similar sections for AWS, GCP, Azure, etc.

The Server docs could link to that section as it's mostly artifact specific but different from all the write permissions here

Reply:
@agilgur5 Should we make a separate issue to address this later?

Reply:
Yea I'll create it